### DISCUSSION TITLE
Uno -WASM => How to use app  service in Uno Client wasm  Am I missing any thing ?=>  Resolved

### DISCUSSION THREAD

**Discussion by NuEdgeInfoTech** [MarimerLLC/csla#2235] [Questions]
#if __WASM__
            Csla.DataPortalClient.HttpProxy.UseTextSerialization = true;
            Csla.Configuration.CslaConfiguration.Configure().
             DataPortal().DefaultProxy(typeof(Csla.DataPortalClient.HttpProxy), @"https://" + "xyz.net" + @"/" + @"api/DataPortalText/");
#else
             Csla.Configuration.CslaConfiguration.Configure().
             DataPortal().DefaultProxy(typeof(Csla.DataPortalClient.HttpProxy), @"https://" + "xyz.net" + @"/" + @"api/DataPortal/");
#endif

 var services = new ServiceCollection();
            services.AddCsla();
            Csla.ApplicationContext.User = new ClaimsPrincipal(new ClaimsIdentity());

![image](https://user-images.githubusercontent.com/76579167/116704960-a5426c80-a9e9-11eb-9d3a-6f7f1c760bb7.png)

**Comment by NuEdgeInfoTech** [comment]
**_principal =new ClaimsPrincipal(new ClaimsIdentity());** after assign this csla api call started working in UNO WASM

```c#
public class ApplicationContextManager : Core.ApplicationContextManager
    {
        private IPrincipal _principal;

        public override IPrincipal GetUser()
        {
            if (_principal == null)
            {
                _principal = new ClaimsPrincipal(new ClaimsIdentity());  // after used this 
                SetUser(_principal);
            }
            return _principal;
        }

        public override void SetUser(IPrincipal principal)
        {
            _principal = principal;
        }
    }
```

**Reply by rockfordlhotka** [reply]
The data portal (by default) will transfer the client-side principal to the server. For this to work, the client-side principal needs to be

1. Not null
2. A serializable type (by MobileFormatter)

I try to automatically set the principal to some valid (unauthorized) value, but that doesn't always happen for various reasons. So sometimes you need to set the value as the app starts up.

In current code, because Microsoft is no longer supporting custom principal types, you should use `ClaimsPrincipal` and `ClaimsIdentity`.
